cmd/kubectl-gadget/trace/trace.go

```diff
@@ -15,16 +15,148 @@
 package trace
 
 import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
 	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
 
+	bindtypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/bindsnoop/types"
+	capabilitiestypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/capabilities/types"
+	dnstypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/dns/types"
+	exectypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/execsnoop/types"
+	fsslowertypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/fsslower/types"
+	mounttypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/mountsnoop/types"
+	oomkilltypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/oomkill/types"
+	opentypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/opensnoop/types"
+	signaltypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/sigsnoop/types"
+	snitypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/snisnoop/types"
+	tcpconnecttypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/tcpconnect/types"
+	tcptypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/tcptracer/types"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+
 	"github.com/spf13/cobra"
 )
 
-// All the gadgets within this package use this global variable, so let's
-// declare it here.
-var params utils.CommonFlags
+// BaseTraceParser is a base for a TraceParser to reuse the shared fields and
+// methods.
+type BaseTraceParser struct {
+	columnsWidth map[string]int
+	outputConfig *utils.OutputConfig
+}
+
+func (p *BaseTraceParser) PrintColumnsHeader(requestedCols []string) {
+	var sb strings.Builder
+
+	for _, col := range requestedCols {
+		if width, ok := p.columnsWidth[col]; ok {
+			sb.WriteString(fmt.Sprintf("%*s", width, strings.ToUpper(col)))
+		}
+
+		// Needed when field is larger than the predefined columnsWidth.
+		sb.WriteRune(' ')
+	}
+
+	fmt.Println(sb.String())
+}
+
+type TraceEvent interface {
+	bindtypes.Event | capabilitiestypes.Event | dnstypes.Event |
+		exectypes.Event | fsslowertypes.Event | mounttypes.Event |
+		oomkilltypes.Event | opentypes.Event | signaltypes.Event |
+		snitypes.Event | tcptypes.Event | tcpconnecttypes.Event
```

Comment:
I was thinking that we could use `any` instead of explicitly listing all types here, however I'm not totally sure about the implications. Do you have any opinion on this?  /cc @alban

Reply:
I know, it doesn't look great. 

I would say that the only advantage of listing them is that we are type-safe. However, I don't expect people to use `TraceEvent` directly but it is something we will only use internally so `any` would be ok as well. I think it is more about how it looks 😄 .

Reply:
Updated with `any`.